sysl2/sysl/ints_view.go

```diff
@@ -0,0 +1,555 @@
+		}
+		v.stringBuilder.WriteString("}\n")
+	}
+
+	return nameMap
+}
+
+func (v *IntsDiagramVisitor) generateStateView(viewParams viewParams, params *IntsParam) string {
+	v.stringBuilder.WriteString("@startuml\n")
+	if viewParams.diagramTitle != "" {
+		fmt.Fprintf(v.stringBuilder, "title %s\n", viewParams.diagramTitle)
+	}
+	v.stringBuilder.WriteString(StateStart)
+	if viewParams.highLightColor != "" {
+		fmt.Fprintf(v.stringBuilder, "  BackgroundColor<<highlight>> %s\n", viewParams.highLightColor)
+	}
+	if viewParams.arrowColor != "" {
+		fmt.Fprintf(v.stringBuilder, "  ArrowColor %s\n", viewParams.arrowColor)
+	}
+
+	if viewParams.indirectArrowColor != "" && viewParams.indirectArrowColor != ArrowColorNone {
+		fmt.Fprintf(v.stringBuilder, "  ArrowColor<<indirect>> %s\n", viewParams.indirectArrowColor)
+	}
+	v.stringBuilder.WriteString("}\n")
+	v.buildClusterForStateView(params.integrations, viewParams.restrictBy)
+	var processed []string
+	for _, dep := range params.integrations {
+		appA := dep.Self.Name
+		appB := dep.Target.Name
+		epA := dep.Self.Endpoint
+		epB := dep.Target.Endpoint
+		_, restrictByAppA := v.mod.Apps[appA].Attrs[viewParams.restrictBy]
+		_, restrictByAppB := v.mod.Apps[appB].Attrs[viewParams.restrictBy]
+		_, restrictByEpA := v.mod.Apps[appA].Endpoints[epA].Attrs[viewParams.restrictBy]
+		_, restrictByEpB := v.mod.Apps[appB].Endpoints[epB].Attrs[viewParams.restrictBy]
+		if viewParams.restrictBy != "" && !restrictByAppA && !restrictByAppB {
+			continue
+		}
+		if viewParams.restrictBy != "" && !restrictByEpA && !restrictByEpB {
+			continue
+		}
+		matchApp := appB
+		matchEp := epB
+		label := ""
+		needsInt := appA != matchApp
+
+		pubSubSrcPtrns := MakeStrSetFromAttr("patterns", v.mod.Apps[appA].Endpoints[epA].Attrs)
+		attrs := map[string]string{}
+		for k, v := range dep.Statement.GetAttrs() {
+			attrs["@"+k] = v.GetS()
+		}
+		var ptrns string
+		var targetPatterns StrSet
+		var srcPtrns StrSet
+		if v.mod.Apps[matchApp].Endpoints[matchEp].Attrs["patterns"] != nil {
+			targetPatterns = MakeStrSetFromAttr("patterns", v.mod.Apps[matchApp].Endpoints[matchEp].Attrs)
+		} else {
+			targetPatterns = MakeStrSetFromAttr("patterns", v.mod.Apps[matchApp].Attrs)
+		}
+		if dep.Statement.GetAttrs()["patterns"] != nil {
+			srcPtrns = MakeStrSetFromAttr("patterns", dep.Statement.GetAttrs())
+		} else {
+			srcPtrns = pubSubSrcPtrns
+		}
+		if srcPtrns != nil || targetPatterns != nil {
+			ptrns = strings.Join(srcPtrns.ToSlice(), ", ") + " → " + strings.Join(targetPatterns.ToSlice(), ", ")
+		} else {
+			ptrns = ""
+		}
+		attrs["patterns"] = ptrns
+		if needsInt {
+			attrs["needs_int"] = strconv.FormatBool(needsInt)
+		}
+		fp := MakeFormatParser(params.app.Attrs["epfmt"].GetS())
+		label = fp.Parse(attrs)
+
+		flow := strings.Join([]string{appA, epB, appB, epB}, ".")
+		isPubSub := v.mod.Apps[appA].Endpoints[epA].GetIsPubsub()
+		epBClient := epB + " client"
+
+		if appA != appB {
+			if isPubSub {
```

Comment:
add test case that generates diagram for `PubSub`

Reply:
added